### PR TITLE
[Fix](nereids) Support partial update for insert into table

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

This PR is a fix for https://github.com/apache/doris/pull/21597

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Review by sohardforaname — APPROVED** [review]

**Comment by bobhan1** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java [hunk 1]
```diff
@@ -191,8 +192,14 @@ public List<Rule> buildRules() {
                             // add cast project
                             List<NamedExpression> castExprs = Lists.newArrayList();
                             for (int i = 0; i < table.getFullSchema().size(); ++i) {
-                                Expression castExpr = TypeCoercionUtils.castIfNotSameType(fullOutputExprs.get(i),
-                                        DataType.fromCatalogType(table.getFullSchema().get(i).getType()));
+                                Column col = table.getFullSchema().get(i);
+                                NamedExpression expr = (NamedExpression) columnToOutput.get(col.getName());
+                                if (expr == null) {
+                                    continue;
+                                }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java:199]: add some comment to explain it

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java [hunk 2]
```diff
@@ -142,6 +141,8 @@ public List<Rule> buildRules() {
                                                     .filter(col -> col.getName().equals(table.getSequenceMapCol()))
                                                     .findFirst().get();
                                             columnToOutput.put(column.getName(), columnToOutput.get(seqCol.getName()));
+                                        } else if (sink.isPartialUpdate()) {
+                                            continue;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java:145]: add some comment to explain it

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p1

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run p0

**Pull request merged**